src/sources/redis/mod.rs

```diff
@@ -226,10 +247,41 @@ async fn handle_line(
 
                 let events = events.into_iter().map(|mut event| {
                     if let Event::Log(ref mut log) = event {
-                        log.try_insert(log_schema().source_type_key(), Bytes::from("redis"));
-                        log.try_insert(log_schema().timestamp_key(), now);
-                        if let Some(redis_key) = redis_key {
-                            event.as_mut_log().insert(redis_key, key);
+                        log_namespace.insert_vector_metadata(
+                            log,
+                            path!(log_schema().source_type_key()),
+                            path!("source_type"),
+                            Bytes::from(RedisSourceConfig::NAME),
+                        );
+                        log_namespace.insert_vector_metadata(
+                            log,
+                            path!(log_schema().timestamp_key()),
+                            path!("timestamp"),
+                            now,
+                        );
+
+                        match log_namespace {
+                            LogNamespace::Vector => log_namespace.insert_source_metadata(
+                                RedisSourceConfig::NAME,
+                                log,
+                                path!("key"),
```

Comment:
Do we need special handling for this here, since the legacy path could be overwritten by the user conf? cc @fuchsnj

Reply:
Yes, unfortunately. I'm going to make a PR improving the helper functions to handle this better (including the manual `match log_namespace` here.
I'll add a link here once that's ready.

Reply:
@fuchsnj is that something we should wait for prior to merging this PR?

Reply:
Preferably, yes, there will be a few changes. It's currently just waiting for CI to pass then it should be merged in.
https://github.com/vectordotdev/vector/pull/15068

Reply:
Preferably, yes, there will be a few changes. It's currently just waiting for CI to pass then it should be merged in.
https://github.com/vectordotdev/vector/pull/15068